File: 2772. Apply Operations to Make All Array Elements Equal to Zero/Solution.py
class Solution:
    def checkArray(self, nums: List[int], k: int) -> bool:
        n, m = len(nums), 0
        for i in range(n):
            if nums[i] - m < 0: return False
            nums[i] -= m
            m += nums[i]
            if i >= k - 1: m -= nums[i - k + 1]
        return m == 0

File: 2772. Apply Operations to Make All Array Elements Equal to Zero/Solution.cpp
class Solution {
public:
    bool checkArray(vector<int>& nums, int k) {
        int n = nums.size(), m = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] - m < 0) return false;
            nums[i] -= m;
            m += nums[i];
            if (i >= k - 1) m -= nums[i - k + 1];
        }
        return m == 0;
    }
};
